mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -39,7 +40,7 @@ import org.apache.spark.sql.DataFrame
 @Experimental
 final class DecisionTreeRegressor(override val uid: String)
   extends Predictor[Vector, DecisionTreeRegressor, DecisionTreeRegressionModel]
-  with DecisionTreeParams with TreeRegressorParams {
+  with TreeRegressorParams {
```

Comment:
Btw, I think I was wrong about this being a breaking API change (because of the reorg you did below), but I still think we should handle this via a DecisionTreeRegressorParams trait.